contrib/dynamic_embedding/README.md

```diff
@@ -0,0 +1,74 @@
+# TorchRec Dynamic Embedding
+
+This folder contains the extension to support dynamic embedding for torchrec. Specifically, this extension enable torchrec to attach an external PS, so that when local GPU embedding is not big enough, we could pull/evict embeddings from/to the PS.
+
+## Installation
+
+After install torchrec, please clone the torchrec repo and manually install the dynamic embedding:
+
+```bash
+git clone git@github.com:pytorch/torchrec.git
+cd contrib/dynamic_embedding
+python setup.py install
+```
+
+And the dynamic embedding will be installed as a separate package named torchrec_dynamic_embedding.
+
+We incorporate `gtest` for the C++ code and use unittest for the python APIs. The tests make sure that the implementation does not have any precision loss. Please turn on the `TDE_WITH_TESTING` in `setup.py` to run tests. Note that for the python test, one needs to set the environment variable `TDE_MEMORY_IO_PATH` to the path of the compiled `memory_io.so`.
+
+## Usage
+
+The dynamic embedding extension has only one api, `tde.wrap`, when wrapping the dataloader and model with it, we will automatically pipeline the data processing and model training. And example of `tde.wrap` is:
+
+```python
+import torchrec_dynamic_embedding as tde
+
+class Model(nn.Module):
+    def __init__(self, config1, config2):
+        super().__init__()
+        self.emb1 = EmbeddingCollection(tables=config1, device=torch.device("meta"))
+        self.emb2 = EmbeddingCollection(tables=config2, device=torch.device("meta"))
+        ...
+
+    def forward(self, kjt1, kjt2):
+        ...
+
+m = Model(config1, config2)
+m = DistributedModelParallel(m)
+dataloader = tde.wrap(
```

Comment:
let's make this into a train_pipeline? (see torchrec/distributed/train_pipelin.py)